### ISSUE TITLE
Questions about classifier

### ISSUE THREAD

**Issue by TingxunShi** [golsun/StyleFusion#2]
Thank you for releasing the project here. After having read your paper and tried the code several times, I've got some questions about the `classifier` you used.

1. First I want to confirm that, classifier is indispensable, right? From the paper I guess you need it to do the inference
2. Suppose I want to train model on my own new style, does it mean that I need to implement the classifier by myself? Do you have the plan to update your code on implementation of the classifier?
3. Seems that the weights of different ngram classifiers are decided by the accuracy of each model. But since style texts are generally far less than dialogue texts, I wonder whether only using accuracy is enough as there would be a potential unbalanced dataset problem. Or when you were training classifiers based on ngram features, did you do some under-sampling to balance the sample amounts of dialogue responses and style texts?
4. I also find that in the ngram.txt, 1-gram items count is roughly equal to the 2/3/4-gram items count. Did you intend to do this, only kept the most frequent items of each category? 

Thanks and look forward to your reply!

**Comment by golsun** [comment]
thanks for your interest!
> 
> 1. First I want to confirm that, classifier is indispensable, right? From the paper I guess you need it to do the inference

yes classifiers are needed to rank the generated hypotheses at inference time.

> 2. Suppose I want to train model on my own new style, does it mean that I need to implement the classifier by myself? Do you have the plan to update your code on implementation of the classifier?

yes you need to train a new classifier. I'll update the [classifier.py](https://github.com/golsun/StyleFusion/blob/master/src/classifier.py) to include the training part.

> 3. Seems that the weights of different ngram classifiers are decided by the accuracy of each model. But since style texts are generally far less than dialogue texts, I wonder whether only using accuracy is enough as there would be a potential unbalanced dataset problem. Or when you were training classifiers based on ngram features, did you do some under-sampling to balance the sample amounts of dialogue responses and style texts?

good question! I trained the model with unbalanced data but tested them with balanced data. and the ngram classifiers are weighted by the testing accuracy on the balanced data.

> 4. I also find that in the ngram.txt, 1-gram items count is roughly equal to the 2/3/4-gram items count. Did you intend to do this, only kept the most frequent items of each category?

yes I only use the most frequent ngrams.

**Comment by TingxunShi** [comment]
Many thanks for the kind reply!

**Comment by LeiShenVictoria** [comment]
Hi, could you please share the classifier training code? Because I want to train the classfier in the my own style context. Many thanks!
